save-backend/src/main/kotlin/com/saveourtool/save/backend/service/ExecutionService.kt

```diff
@@ -61,13 +61,14 @@ class ExecutionService(
     }
 
     /**
-     * @param execution [Execution]
+     * @param srcExecution [Execution]
      * @param newStatus [Execution.status]
      * @throws ResponseStatusException
      */
     @Transactional
-    fun updateExecutionStatus(execution: Execution, newStatus: ExecutionStatus) {
-        log.debug("Updating status to $newStatus on execution id = ${execution.requiredId()}")
+    fun updateExecutionStatus(srcExecution: Execution, newStatus: ExecutionStatus) {
+        log.debug("Updating status to $newStatus on execution id = ${srcExecution.requiredId()}")
+        val execution = executionRepository.findWithLockingById(srcExecution.requiredId()).orElse(null).orNotFound()
```

Comment:
I suggest to make a version of `orNotFound` with `Optional` as a receiver

Reply:
or change `findWithLockingById` to return `Execution?`

Reply:
changed to `Execution?`